flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamConfig.java

```diff
@@ -132,6 +136,12 @@ public class StreamConfig implements Serializable {
 
     private final Configuration config;
 
+    private final transient Map<String, Object> objectSerializationConfig = new HashMap<>();
+    private final transient Map<Integer, CompletableFuture<StreamConfig>> chainedTaskFutures =
+            new HashMap<>();
+    private final transient CompletableFuture<StreamConfig> serializationFuture =
+            new CompletableFuture<>();
```

Comment:
I'd suggest to add a brief explanation here:
- We first collect all the need-to-be-serialized objects, then serialize them at once
- The purpose is to make parallelization of `StreamConfig` serialization easier.